service LampService {
    rpc SetLamp(Lamp) returns (Status) {}
    rpc GetLamp(Empty) returns (Lamp) {}
}

service AirConditionerService {
    rpc SetAirConditioner(AirConditioner) returns (Status) {}
    rpc GetAirConditioner(Empty) returns (AirConditioner) {}
}

service AudioSystemService {
    rpc SetAudioSystem(AudioSystem) returns (Status) {}
    rpc GetAudioSystem(Empty) returns (AudioSystem) {}
}

message Lamp {
    required bool on = 1;
}

message AirConditioner {
    required bool on = 1;
    required float temperature = 2;
}

message AudioSystem {
    required bool on = 1;
    required string current_song = 2;
}

message Status {
    required int32 status = 1;
}

message Empty {}